server/netty-server/src/main/scala/sttp/tapir/server/netty/internal/NettyServerHandler.scala

```diff
@@ -40,6 +47,40 @@ class NettyServerHandler[F[_]](route: Route[F], unsafeRunAsync: (() => F[Unit])
     res
   }
 
+  override def handlerAdded(ctx: ChannelHandlerContext): Unit =
+    if (ctx.channel.isActive) {
+      initHandler(ctx)
+    }
+  override def channelActive(ctx: ChannelHandlerContext): Unit = initHandler(ctx)
+
+  override def channelReadComplete(ctx: ChannelHandlerContext): Unit = {
+    logger.trace(s"channelReadComplete: ctx = $ctx")
+    // The normal response to read complete is to issue another read,
```

Comment:
hm is that needed for the `pendingResponses` logic?

Reply:
These ctx.read() calls are IMO unnecessary. For some reason http4s does AUTO_READ=false, but I don't see this interferring with our implementation. Reactive streams also work as expected.